core/aws_ddk_core/cicd/pipeline.py

```diff
@@ -243,10 +247,94 @@ def add_stage(
         manual_approvals = manual_approvals or self._config.get_env_config(stage_id).get("manual_approvals")
 
         self._pipeline.add_stage(
-            stage, pre=[ManualApprovalStep(f"PromoteTo{stage_id.title()}")] if manual_approvals else None
+            stage,
+            pre=[ManualApprovalStep(f"PromoteTo{stage_id.title()}")] if manual_approvals else None,
+        )
+        return self
+
+    def add_security_lint_stage(
+        self,
+        stage_name: Optional[str] = None,
+        cloud_assembly_file_set: Optional[IFileSetProducer] = None,
+    ) -> "CICDPipelineStack":
+        """
+        Add linting - cfn-nag, and bandit.
+
+        Parameters
+        ----------
+        stage_name: Optional[str]
+            Name of the stage
+        cloud_assembly_file_set: Optional[IFileSetProducer]
+            Cloud assembly file set producer
+
+        Returns
+        -------
+        pipeline : CICDPipeline
+            CICD pipeline
+        """
+        self._pipeline.add_wave(
+            stage_name or "SecurityLint",
+            post=[
+                get_cfn_nag_action(
+                    file_set_producer=cloud_assembly_file_set
+                    if cloud_assembly_file_set
+                    else self._pipeline.cloud_assembly_file_set
+                ),
+                get_bandit_action(code_pipeline_source=self._source_action),
+            ],
+        )
+        return self
+
+    def add_test_stage(
+        self,
+        stage_name: Optional[str] = None,
+        cloud_assembly_file_set: Optional[IFileSetProducer] = None,
+        commands: Optional[List[str]] = None,
+    ) -> "CICDPipelineStack":
+        """
+        Add test - e.g. pytest.
+
+        Parameters
+        ----------
+        stage_name: Optional[str]
+            Name of the stage
+        cloud_assembly_file_set: Optional[IFileSetProducer]
+            Cloud assembly file set
+        commands: Optional[List[str]]
+            Additional commands to run in the test. Defaults to "./test.sh" otherwise
+
+        Returns
+        -------
+        pipeline : CICDPipelineStack
+            CICD pipeline
+        """
+
+        self._pipeline.add_wave(
+            stage_name or "Tests",
+            post=[
+                get_tests_action(
+                    file_set_producer=cloud_assembly_file_set if cloud_assembly_file_set else self._source_action,
+                    commands=commands,
+                ),
+            ],
         )
         return self
 
+    def add_checks(self) -> "CICDPipelineStack":
+        """
+        Add checks to the pipeline (e.g. linting, security, tests...).
+
+        Returns
+        -------
+        pipeline : CICDPipelineStack
+            CICD pipeline
+        """
+        if self._config.get_env_config("cicd").get("execute_security_lint"):
```

Comment:
This is assuming that the name of the `environment_id` is `cicd`. We used to assume that is the case in Orion, but now the name of the CICD stage is up to the user. Instead, we should `self.environment_id`

Same applied to `execute_tests` below

Reply:
resolved in: https://github.com/awslabs/aws-ddk/pull/44